classloader-leak-prevention/classloader-leak-prevention-core/src/main/java/se/jiderhamn/classloader/leak/prevention/cleanup/MXBeanNotificationListenersCleanUp.java

```diff
@@ -18,72 +21,53 @@
-                      final NotificationListener listener = preventor.getFieldValue(listenerField, listenerInfo);
-                      final NotificationListener rawListener = unwrap(preventor, listenerWrapperClass, listener);
-                      final NotificationFilter filter = preventor.getFieldValue(filterField, listenerInfo);
-                      final Object handback = preventor.getFieldValue(handbackField, listenerInfo);
+    if (platformInterfaces != null) {
+      for (Class<? extends PlatformManagedObject> platformInterface : platformInterfaces) {
+        for (Object mxBean : ManagementFactory.getPlatformMXBeans(platformInterface)) {
+          if (mxBean instanceof NotificationEmitter) { // The MXBean may have NotificationListeners
+            if (canProcessNotificationEmitterSupport && notificationEmitterSupportClass.isAssignableFrom(mxBean.getClass())) {
+              final List<? /* NotificationEmitterSupport.ListenerInfo */> listenerList = preventor.getFieldValue(listenerListField, mxBean);
+              if (listenerList != null) {
+                for (Object listenerInfo : listenerList) { // Loop all listeners
+                  final NotificationListener listener = preventor.getFieldValue(listenerField, listenerInfo);
+                  final NotificationListener rawListener = unwrap(preventor, listenerWrapperClass, listener);
+                  final NotificationFilter filter = preventor.getFieldValue(filterField, listenerInfo);
+                  final Object handback = preventor.getFieldValue(handbackField, listenerInfo);
 
-                      if(preventor.isLoadedInClassLoader(rawListener) || preventor.isLoadedInClassLoader(filter) || preventor.isLoadedInClassLoader(handback)) {
-                        preventor.warn(((listener == rawListener) ? "Listener '" : "Wrapped listener '") + listener + 
-                            "' (or its filter or handback) of MXBean " + mxBean + 
-                            " of PlatformComponent " + platformComponent + " was loaded in protected ClassLoader; removing");
-                        // This is safe, as the implementation (as of this writing) works with a copy, not altering the original
-                        try {
-                          ((NotificationEmitter) mxBean).removeNotificationListener(listener, filter, handback);
-                        }
-                        catch (ListenerNotFoundException e) { // Should never happen
-                          preventor.error(e);
-                        }
-                      }
+                  if (preventor.isLoadedInClassLoader(rawListener) || preventor.isLoadedInClassLoader(filter) || preventor.isLoadedInClassLoader(handback)) {
+                    preventor.warn(((listener == rawListener) ? "Listener '" : "Wrapped listener '") + listener + "' (or its filter or handback) of MXBean " + mxBean + " of PlatformManagedObject " + platformInterface + " was loaded in protected ClassLoader; removing");
```

Comment:
Would you care to re-add some linebreaks in here...?
```suggestion
                    preventor.warn(((listener == rawListener) ? "Listener '" : "Wrapped listener '") + listener + 
                    "' (or its filter or handback) of MXBean " + mxBean + 
                    " of PlatformManagedObject " + platformInterface + " was loaded in protected ClassLoader; removing");
```